airflow-core/src/airflow/configuration.py

```diff
@@ -347,6 +347,7 @@ def sensitive_config_values(self) -> set[tuple[str, str]]:
     # DeprecationWarning will be issued and the old option will be used instead
     deprecated_options: dict[tuple[str, str], tuple[str, str, str]] = {
         ("dag_processor", "refresh_interval"): ("scheduler", "dag_dir_list_interval", "3.0"),
+        ("dag_processor", "parsing_pre_import_modules"): ("scheduler", "parsing_pre_import_modules", "3.0.2"),
```

Comment:
```suggestion
        ("dag_processor", "parsing_pre_import_modules"): ("scheduler", "parsing_pre_import_modules", "3.1.0"),
```
Let's plan this for 3.1.0 since it's not a bug fix but an improvement